src/Core/Resolvers/MultipleCreateOrderHelper.cs

```diff
@@ -0,0 +1,340 @@
+                if (doesSourceContainAnyAutogenRelationshipField && doesTargetContainAnyAutogenRelationshipField)
+                {
+                    throw new DataApiBuilderException(
+                        message: $"Both source entity: {sourceEntityName} and target entity: {targetEntityName} contain autogenerated fields for relationship: {relationshipName} at level: {nestingLevel}",
+                        statusCode: HttpStatusCode.BadRequest,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+                }
+
+                // If the source entity contains a relationship field in request body which suggests us to do the insertion first in source entity,
+                // and there is an autogenerated relationship field in the target entity which suggests us to do insertion first in target entity,
+                // we cannot determine a valid order of insertion.
+                if (doesSourceBodyContainAnyRelationshipField && doesTargetContainAnyAutogenRelationshipField)
+                {
+                    throw new DataApiBuilderException(
+                        message: $"Input for source entity: {sourceEntityName} for the relationship: {relationshipName} at level: {nestingLevel} cannot contain the field: {relationshipFieldInSource}.",
+                        statusCode: HttpStatusCode.BadRequest,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+                }
+
+                // If the target entity contains a relationship field in request body which suggests us to do the insertion first in target entity,
+                // and there is an autogenerated relationship field in the source entity which suggests us to do insertion first in source entity,
+                // we cannot determine a valid order of insertion.
+                if (doesTargetBodyContainAnyRelationshipField && doesSourceContainAnyAutogenRelationshipField)
+                {
+                    throw new DataApiBuilderException(
+                        message: $"Input for target entity: {targetEntityName} for the relationship: {relationshipName} at level: {nestingLevel} cannot contain the field: {relationshipFieldInSource}.",
+                        statusCode: HttpStatusCode.BadRequest,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+                }
+
+                // Determine whether the input data for source/target contain a value (could be null) for this pair of relationship fields. 
+                bool doesSourceBodyContainThisRelationshipColumn = columnDataInSourceBody.TryGetValue(relationshipFieldInSource, out IValueNode? sourceColumnvalue);
+                bool doesTargetBodyContainThisRelationshipColumn = columnDataInTargetBody.TryGetValue(relationshipFieldInTarget, out IValueNode? targetColumnvalue);
```

Comment:
Since there are variables representing aggregrate booleans for all the columns like `doesSourceBodyContainAnyRelationshipField`/`canSourceAssumeAllRelationshipFieldValues`, I preferred to add `this` in the variable name to make it clear that using this boolean, we are just talking about the current relationship column.